### PR TITLE
SNOW-1347052 Update pandas API PuPr warning messages

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1347052 Update pandas API PuPr warning messages

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

- change the words from "private preview" to "public preview"

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/__init__.py [hunk 1]
```diff
@@ -46,8 +46,8 @@
 
 
 warnings.warn(
-    "Snowpark pandas is currently in Private Preview."
-    + " See https://docs.snowflake.com/LIMITEDACCESS/snowpark-pandas for details.",
+    "Snowpark pandas is in Public Preview since 1.17.0."
```
Comment (sfc-gh-vbudati) [inline src/snowflake/snowpark/modin/plugin/__init__.py:49]:
```suggestion
    "Snowpark pandas has been in Public Preview since 1.17.0."
```

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM with Varnika's suggested change

File: src/snowflake/snowpark/modin/plugin/__init__.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/__init__.py:49]: +1

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Except for @sfc-gh-vbudati's nit, LGTM!

**Review by sfc-gh-vbudati — APPROVED** [review]

**Pull request merged**